### ISSUE TITLE
Arabic Translation Progress

### ISSUE THREAD

**Issue by iliakan** [javascript-tutorial/ar.javascript.info#1]
## Maintainer List

@alahnomi

## For New Translators

<details><summary><b>Please read this first (click to open)</b></summary>
<p>
To translate an article:

1. Check that no one else has claimed your article in the checklist below.
2. Comment below with the title of the article that you would like to translate, e.g. `An Introduction to JavaScript`. 
    - **Please take only one article at a time**.
3. Fork this repo, translate the article in your fork and submit a pull request!
    - The pull request title should be same as the article, e.g. `An Introduction to JavaScript` (just like comment)

Please be prompt with your translations! If you find find that you can't commit any more, let the maintainers know so they can assign the page to someone else.
</p>
</details>

## For Maintainers

<details><summary><b>Click to open</b></summary>
<p>
Please let others know what you do, on community boards and chats, invite them to join. Translations become better if more people see them.

For teams we suggest that an article has 2 reviews to be merged. 

Translalions are tracked below, like this:

* [ ] [Home Page](url) (@iliakan) #1

There's a helper bot to track translations, you can read more about it at <https://github.com/javascript-tutorial/translate/edit/master/BOT.md>.

If something doesn't work right, please contact @iliakan. 

</p>
</details>

Only maintainers can check/uncheck items below. If you're not, please write in a comment what you take to translate.

### An introduction

* [ ] [An Introduction to JavaScript](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/01-getting-started/1-intro)
* [ ] [Manuals and specifications](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/01-getting-started/2-manuals-specifications)
* [ ] [Code editors](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/01-getting-started/3-code-editors)
* [ ] [Developer console](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/01-getting-started/4-devtools)

### JavaScript Fundamentals

* [ ] [Hello, world!](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/01-hello-world)
* [ ] [Code structure](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/02-structure)
* [ ] [The modern mode, "use strict"](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/03-strict-mode)
* [ ] [Variables](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/04-variables)
* [ ] [Data types](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/05-types)
* [ ] [Type Conversions](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/06-type-conversions)
* [ ] [Operators](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/07-operators)
* [ ] [Comparisons](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/08-comparison)
* [ ] [Interaction: alert, prompt, confirm](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/09-alert-prompt-confirm)
* [ ] [Conditional operators: if, '?'](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/10-ifelse)
* [ ] [Logical operators](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/11-logical-operators)
* [ ] [Loops: while and for](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/12-while-for)
* [ ] [The "switch" statement](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/13-switch)
* [ ] [Functions](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/14-function-basics)
* [ ] [Function expressions and arrows](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/15-function-expressions-arrows)
* [ ] [JavaScript specials](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/02-first-steps/16-javascript-specials)

### Code quality

* [ ] [Debugging in Chrome](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/03-code-quality/01-debugging-chrome)
* [ ] [Coding Style](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/03-code-quality/02-coding-style)
* [ ] [Comments](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/03-code-quality/03-comments)
* [ ] [Ninja code](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/03-code-quality/04-ninja-code)
* [ ] [Automated testing with mocha](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/03-code-quality/05-testing-mocha)
* [ ] [Polyfills](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/03-code-quality/06-polyfills)

### Objects: the basics

* [ ] [Objects](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/04-object-basics/01-object)
* [ ] [Garbage collection](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/04-object-basics/02-garbage-collection)
* [ ] [Symbol type](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/04-object-basics/03-symbol)
* [ ] [Object methods, "this"](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/04-object-basics/04-object-methods)
* [ ] [Object to primitive conversion](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/04-object-basics/05-object-toprimitive)
* [ ] [Constructor, operator "new"](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/04-object-basics/06-constructor-new)

### Data types

* [ ] [Methods of primitives](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/05-data-types/01-primitives-methods)
* [ ] [Numbers](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/05-data-types/02-number)
* [ ] [Strings](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/05-data-types/03-string)
* [ ] [Arrays](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/05-data-types/04-array)
* [ ] [Array methods](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/05-data-types/05-array-methods)
* [ ] [Iterables](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/05-data-types/06-iterable)
* [ ] [Map and Set](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/05-data-types/07-map-set)
* [ ] [WeakMap and WeakSet](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/05-data-types/08-weakmap-weakset)
* [ ] [Object.keys, values, entries](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/05-data-types/09-keys-values-entries)
* [ ] [Destructuring assignment](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/05-data-types/10-destructuring-assignment)
* [ ] [Date and time](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/05-data-types/11-date)
* [ ] [JSON methods, toJSON](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/05-data-types/12-json)

### Advanced working with functions

* [ ] [Recursion and stack](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/06-advanced-functions/01-recursion)
* [ ] [Rest parameters and spread operator](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/06-advanced-functions/02-rest-parameters-spread-operator)
* [ ] [Closure](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/06-advanced-functions/03-closure)
* [ ] [The old "var"](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/06-advanced-functions/04-var)
* [ ] [Global object](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/06-advanced-functions/05-global-object)
* [ ] [Function object, NFE](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/06-advanced-functions/06-function-object)
* [ ] [The "new Function" syntax](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/06-advanced-functions/07-new-function)
* [ ] [Scheduling: setTimeout and setInterval](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/06-advanced-functions/08-settimeout-setinterval)
* [ ] [Decorators and forwarding, call/apply](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/06-advanced-functions/09-call-apply-decorators)
* [ ] [Function binding](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/06-advanced-functions/10-bind)
* [ ] [Arrow functions revisited](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/06-advanced-functions/12-arrow-functions)

### Object properties configuration

* [ ] [Property flags and descriptors](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/07-object-properties/01-property-descriptors)
* [ ] [Property getters and setters](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/07-object-properties/02-property-accessors)

### Prototypes, inheritance

* [ ] [Prototypal inheritance](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/08-prototypes/01-prototype-inheritance)
* [ ] [F.prototype](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/08-prototypes/02-function-prototype)
* [ ] [Native prototypes](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/08-prototypes/03-native-prototypes)
* [ ] [Prototype methods, objects without __proto__](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/08-prototypes/04-prototype-methods)

### Classes

* [ ] [Class basic syntax](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/09-classes/01-class)
* [ ] [Class inheritance](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/09-classes/02-class-inheritance)
* [ ] [Static properties and methods](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/09-classes/03-static-properties-methods)
* [ ] [Private and protected properties and methods](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/09-classes/04-private-protected-properties-methods)
* [ ] [Extending built-in classes](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/09-classes/05-extend-natives)
* [ ] [Class checking: "instanceof"](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/09-classes/06-instanceof)
* [ ] [Mixins](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/09-classes/07-mixins)

### Error handling

* [ ] [Error handling, "try..catch"](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/10-error-handling/1-try-catch)
* [ ] [Custom errors, extending Error](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/10-error-handling/2-custom-errors)

### Promises, async/await

* [ ] [Introduction: callbacks](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/11-async/01-callbacks)
* [ ] [Promise](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/11-async/02-promise-basics)
* [ ] [Promises chaining](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/11-async/03-promise-chaining)
* [ ] [Error handling with promises](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/11-async/04-promise-error-handling)
* [ ] [Promise API](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/11-async/05-promise-api)
* [ ] [Promisification](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/11-async/06-promisify)
* [ ] [Microtasks](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/11-async/07-microtask-queue)
* [ ] [Async/await](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/11-async/08-async-await)

### Generators, advanced iteration

* [ ] [Generators](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/12-generators-iterators/1-generators)
* [ ] [Async iterators and generators](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/12-generators-iterators/2-async-iterators-generators)

### Modules

* [ ] [Modules, introduction](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/13-modules/01-modules-intro)
* [ ] [Export and Import](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/13-modules/02-import-export)
* [ ] [Dynamic imports](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/13-modules/03-modules-dynamic-imports)

### Miscellaneous

* [ ] [Proxy and Reflect](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/99-js-misc/01-proxy)
* [ ] [Eval: run a code string](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/99-js-misc/02-eval)
* [ ] [Currying](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/1-js/99-js-misc/03-currying-partials)

### Document

* [ ] [Browser environment, specs](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/1-document/01-browser-environment)
* [ ] [DOM tree](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/1-document/02-dom-nodes)
* [ ] [Walking the DOM](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/1-document/03-dom-navigation)
* [ ] [Searching: getElement*, querySelector*](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/1-document/04-searching-elements-dom)
* [ ] [Node properties: type, tag and contents](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/1-document/05-basic-dom-node-properties)
* [ ] [Attributes and properties](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/1-document/06-dom-attributes-and-properties)
* [ ] [Modifying the document](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/1-document/07-modifying-document)
* [ ] [Styles and classes](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/1-document/08-styles-and-classes)
* [ ] [Element size and scrolling](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/1-document/09-size-and-scroll)
* [ ] [Window sizes and scrolling](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/1-document/10-size-and-scroll-window)
* [ ] [Coordinates](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/1-document/11-coordinates)

### Introduction to Events

* [ ] [Introduction to browser events](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/2-events/01-introduction-browser-events)
* [ ] [Bubbling and capturing](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/2-events/02-bubbling-and-capturing)
* [ ] [Event delegation](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/2-events/03-event-delegation)
* [ ] [Browser default actions](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/2-events/04-default-browser-action)
* [ ] [Dispatching custom events](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/2-events/05-dispatch-events)

### UI Events

* [ ] [Mouse events basics](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/3-event-details/1-mouse-events-basics)
* [ ] [Moving: mouseover/out, mouseenter/leave](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/3-event-details/3-mousemove-mouseover-mouseout-mouseenter-mouseleave)
* [ ] [Drag'n'Drop with mouse events](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/3-event-details/4-mouse-drag-and-drop)
* [ ] [Keyboard: keydown and keyup](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/3-event-details/5-keyboard-events)
* [ ] [Scrolling](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/3-event-details/8-onscroll)

### Forms, controls

* [ ] [Form properties and methods](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/4-forms-controls/1-form-elements)
* [ ] [Focusing: focus/blur](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/4-forms-controls/2-focus-blur)
* [ ] [Events: change, input, cut, copy, paste](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/4-forms-controls/3-events-change-input)
* [ ] [Forms: event and method submit](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/4-forms-controls/4-forms-submit)

### Document and resource loading

* [ ] [Page: DOMContentLoaded, load, beforeunload, unload](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/5-loading/01-onload-ondomcontentloaded)
* [ ] [Scripts: async, defer](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/5-loading/02-script-async-defer)
* [ ] [Resource loading: onload and onerror](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/5-loading/03-onload-onerror)

### Miscellaneous

* [ ] [Mutation observer](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/99-ui-misc/01-mutation-observer)
* [ ] [Selection and Range](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/99-ui-misc/02-selection-range)
* [ ] [Event loop: microtasks and macrotasks](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/2-ui/99-ui-misc/03-event-loop)

## Frames and windows

* [ ] [Popups and window methods](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/3-frames-and-windows/01-popup-windows)
* [ ] [Cross-window communication](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/3-frames-and-windows/03-cross-window-communication)
* [ ] [The clickjacking attack](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/3-frames-and-windows/06-clickjacking)

## Binary data, files

* [ ] [ArrayBuffer, binary arrays](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/4-binary/01-arraybuffer-binary-arrays)
* [ ] [TextDecoder and TextEncoder](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/4-binary/02-text-decoder)
* [ ] [Blob](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/4-binary/03-blob)
* [ ] [File and FileReader](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/4-binary/04-file)

## Network requests

* [ ] [Fetch](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/5-network/01-fetch)
* [ ] [FormData](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/5-network/02-formdata)
* [ ] [Fetch: Download progress](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/5-network/03-fetch-progress)
* [ ] [Fetch: Abort](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/5-network/04-fetch-abort)
* [ ] [Fetch: Cross-Origin Requests](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/5-network/05-fetch-crossorigin)
* [ ] [Fetch API](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/5-network/06-fetch-api)
* [ ] [URL objects](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/5-network/07-url)
* [ ] [XMLHttpRequest](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/5-network/08-xmlhttprequest)
* [ ] [Resumable file upload](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/5-network/09-resume-upload)
* [ ] [Long polling](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/5-network/10-long-polling)
* [ ] [WebSocket](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/5-network/11-websocket)
* [ ] [Server Sent Events](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/5-network/12-server-sent-events)

## Storing data in the browser

* [ ] [Cookies, document.cookie](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/6-data-storage/01-cookie)
* [ ] [LocalStorage, sessionStorage](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/6-data-storage/02-localstorage)
* [ ] [IndexedDB](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/6-data-storage/03-indexeddb)

## Animation

* [ ] [Bezier curve](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/7-animation/1-bezier-curve)
* [ ] [CSS-animations](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/7-animation/2-css-animations)
* [ ] [JavaScript animations](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/7-animation/3-js-animation)

## Web components

* [ ] [From the orbital height](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/8-web-components/1-webcomponents-intro)
* [ ] [Custom elements](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/8-web-components/2-custom-elements)
* [ ] [Shadow DOM](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/8-web-components/3-shadow-dom)
* [ ] [Template element](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/8-web-components/4-template-element)
* [ ] [Shadow DOM slots, composition](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/8-web-components/5-slots-composition)
* [ ] [Shadow DOM styling](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/8-web-components/6-shadow-dom-style)
* [ ] [Shadow DOM and events](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/8-web-components/7-shadow-dom-events)

## Regular expressions

* [ ] [Patterns and flags](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/01-regexp-introduction)
* [ ] [Methods of RegExp and String](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/02-regexp-methods)
* [ ] [Character classes](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/03-regexp-character-classes)
* [ ] [Escaping, special characters](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/04-regexp-escaping)
* [ ] [Sets and ranges [...]](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/05-regexp-character-sets-and-ranges)
* [ ] [Quantifiers +, *, ? and {n}](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/07-regexp-quantifiers)
* [ ] [Greedy and lazy quantifiers](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/08-regexp-greedy-and-lazy)
* [ ] [Capturing groups](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/09-regexp-groups)
* [ ] [Backreferences in pattern: \n and \k](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/10-regexp-backreferences)
* [ ] [Alternation (OR) |](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/11-regexp-alternation)
* [ ] [String start ^ and finish $](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/12-regexp-anchors)
* [ ] [Multiline mode, flag "m"](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/13-regexp-multiline-mode)
* [ ] [Lookahead and lookbehind](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/14-regexp-lookahead-lookbehind)
* [ ] [Infinite backtracking problem](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/15-regexp-infinite-backtracking-problem)
* [ ] [Unicode: flag "u"](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/20-regexp-unicode)
* [ ] [Unicode character properties \p](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/21-regexp-unicode-properties)
* [ ] [Sticky flag "y", searching at position](https://github.com/javascript-tutorial/ar.javascript.info/blob/master/9-regular-expressions/22-regexp-sticky)

**Comment by musaab-abdalla** [comment]
Hello @iliakan I would like to join the Arabic translation team, please add me so I can start do some translation process.

**Comment by iliakan** [comment]
You can submit PR right now, there's no need to become a maintainer for that =). See the top of this issue.

If your PRs are good, we'll add you as a maintainer.

**Comment by HNBR** [comment]
Debugging in Chrome

**Comment by Emmanuel-Melon** [comment]
Hey there, I'd love to take part in translating this awesome project.
Can I take on the `Objects` article?

**Comment by iliakan** [comment]
@alahnomi, @Jamil-Bailony , are you online?

**Comment by saadishahin** [comment]
Hello,
I would like to boost up the translation as i would like to see this site offered in Arabic, i´ll ask for more contributors.

i´ll start with ¨**Function Expressions**¨

**Comment by iliakan** [comment]
If you have enough JS knowledge, I can make you a contributor.

**Comment by LeOndaz** [comment]
Ninja code

**Comment by OmarioHasan** [comment]
@iliakan  
Hello World <3 
I can give help with this amazing website 
As I'm the founder of Angular Army 'Ng Arab' I can help with my team for the whole translation 
Just let me know what's done and what's not and how many developers we need  
Regards

**Comment by iliakan** [comment]
Hi @OmarioHasan,

Guess, you can take a look at the existing PRs and this repo to see what's done =)

The tutorial is big, so there's a lot to translate, because modern JS has a lot of features =)

**Comment by OmarioHasan** [comment]
@iliakan I've already taken a look but didn't find any Arabic words in this PRs

**Comment by iliakan** [comment]
@OmarioHasan https://github.com/javascript-tutorial/ar.javascript.info/pull/2/files this isn't Arabic?

**Comment by OmarioHasan** [comment]
@iliakan  yes sure 
didn't this translation merged yet?

**Comment by iliakan** [comment]
@OmarioHasan I don't see much activity in Arabic.

You want to take it over?

**Comment by iliakan** [comment]
@alahnomi, @Jamil-Bailony is there anything you've fully translted?

**Comment by OmarioHasan** [comment]
> @OmarioHasan I don't see much activity in Arabic.
> 
> You want to take it over?

I just need to help because this website is very valuable
Any kind of help that I can support then I'm in

**Comment by iliakan** [comment]
@OmarioHasan Do you think Arabic version is needed? I'm not familiar with the state of the web development in Arabic countries.

**Comment by OmarioHasan** [comment]
@iliakan 
You are right we have a weakness in many points but the current generation may change this weakness.
here you can find Ng Arab Community: https://www.angular-communities.com It's my first step to help Arabian developers.
If you need to fill this branch with Arabic words I can support you and thanks for your efforts for spreading the knowledge 
Regards

**Comment by iliakan** [comment]
@OmarioHasan I'm helping to translate the tutorial in languages that are much less popular than Arabic. 

So what we can do is to take the current English version and translate it, and maybe make use of the existing PR.

@OmarioHasan May I assume that you're familiar with git?

**Comment by OmarioHasan** [comment]
@iliakan  yes sure

**Comment by iliakan** [comment]
@OmarioHasan All right, it's done! You're the contributor now (got an invite?)

I also updated the repo from upstream.

**Comment by OmarioHasan** [comment]
@iliakan got it thanks
I assume we will translate all topics 
I mean there is no pending on in progress translate for any topic is that right

**Comment by iliakan** [comment]
I only see one PR.

**Comment by OmarioHasan** [comment]
@iliakan  
Our heroes listed [here ](https://docs.google.com/spreadsheets/d/1axChrzLm-DLpAG9OjoWzBz9hu0sfxttlQ9GyhsegZ4E/edit?fbclid=IwAR1xgtwgwxg7vNQPcdQqyfhGWNYlRyxbnNVb1I-DEEVREhQ5DjOadC0k5rQ#gid=0)
When the team is ready, we will start right away

**Comment by iliakan** [comment]
@OmarioHasan what stops them now?

**Comment by iliakan** [comment]
@OmarioHasan did you read the issue top? https://github.com/javascript-tutorial/ar.javascript.info/issues/1 

There are some details about the translation bot there. It helps to facilitate team translation.

**Comment by OmarioHasan** [comment]
@iliakan 
We will make a slack to discuss what we will do
If you need to join and say something to the team it will be fine

**Comment by OmarioHasan** [comment]
@iliakan yes I've read the full details is there something I need to know other ?

**Comment by iliakan** [comment]
@OmarioHasan is there anything else I can do here?

**Comment by iliakan** [comment]
@OmarioHasan everything's set up for the translation.

**Comment by OmarioHasan** [comment]
> @OmarioHasan is there anything else I can do here?

No thank you

**Comment by OmarioHasan** [comment]
@iliakan Hello
I've a question about the direction
It must be RTL in Arabic
Is there a something we can make now during the translation to help the direction or just let it for now ?

**Comment by iliakan** [comment]
@OmarioHasan Can you install the tutorial server locally? And see how it goes https://github.com/javascript-tutorial/server

**Comment by iliakan** [comment]
I can apply necessary fixed to styles, if any required.

**Comment by OmarioHasan** [comment]
@iliakan I used the server already but just asking about if we can fix the style or not 
And can we fix it now or after translation complete.
That was a starter lesson for Ng Arab community to start contribute to javascriot.info : https://youtu.be/pq8Dnmkrnuk

**Comment by iliakan** [comment]
@OmarioHasan of course we can fix the styles, you can fix them as appropriate and submit a PR. Just mark what's ltr-only and maybe Arabic-only.

**Comment by OmarioHasan** [comment]
@iliakan  
I think if i make a changes for this file `body.styl` it will affect the whole project
![image](https://user-images.githubusercontent.com/8146344/82571472-4a6e8700-9b83-11ea-937b-af212daceee9.png)
That's what i need to do for the view so it will be kind of you if you can guide me to steps for AR style
![image](https://user-images.githubusercontent.com/8146344/82571672-93264000-9b83-11ea-990a-4183b4ec3efd.png)

**Comment by OmarioHasan** [comment]
@iliakan  can we have a direct chat better than commenting to make sure all the process the working well with us

**Comment by iliakan** [comment]
@OmarioHasan We can, but I'm not a specialist in RTL-sites. All I did was LTR.

I hope there's a way to properly fix styles =)

**Comment by iliakan** [comment]
By the way, you can apply styles for Arabic-only with `if`.

Like here for Japanese:

```
if lang=='ja'
    font = BlinkMacSystemFont, -apple-system, "Hiragino Kaku Gothic ProN","メイリオ", sans-serif
  else
    font = BlinkMacSystemFont, -apple-system, Segoe UI, Roboto, Helvetica, Arial, sans-serif
serif
```

**Comment by OmarioHasan** [comment]
this styles to include inside `.styl` extension is that right ?

**Comment by iliakan** [comment]
Right.

**Comment by iliakan** [comment]
Stylus preprocessor

**Comment by OmarioHasan** [comment]
@iliakan  Sorry for too much asking but i think this is not working 
is there a something related to locales missed for AR Version ? 
![image](https://user-images.githubusercontent.com/8146344/82582560-907f1700-9b92-11ea-931b-03f181520151.png)

**Comment by iliakan** [comment]
Well, `rtl.styl` was created for Fahrsi language (also rtl), we can use for `ar` also, of course!

Is it ok that the sidebar remains on the left in Arabic?

**Comment by iliakan** [comment]
p.s. not sure if `rtl.styl` is used now, I created it just to make sure things can work out.

**Comment by OmarioHasan** [comment]
Sidebar must be on the right 
I'm tried to specify the style for ar in body.styl and rtl.style but seems not to work in both

**Comment by iliakan** [comment]
You'd like to have a working `rtl.styl`? Anything else?

**Comment by OmarioHasan** [comment]
No thank you and thanks for your help 😊

**Comment by iliakan** [comment]
No - means no `rtl.styl`? Nothing needed? 😮

**Comment by OmarioHasan** [comment]
No I just need only a working rtl.styl

**Comment by iliakan** [comment]
Ok, I'll do it and write here.

**Comment by iliakan** [comment]
I pushed the change. Instead of a separate `rtl.styl` file, you have the variable `rtl` in styles now.

That's better, as RTL styles can be naturally embedded into stylesheets.

Like this:

```
body
  color white
  if rtl
    direction rtl
```

**Comment by OmarioHasan** [comment]
Got it thanks
I will make a necessary changes and then make a pull request

**Comment by OmarioHasan** [comment]
@iliakan  
For sorry your latest changes didn't work with me
![image](https://user-images.githubusercontent.com/8146344/82606720-d6020b00-9bb7-11ea-9dbc-cc80aeac4f19.png)
![image](https://user-images.githubusercontent.com/8146344/82606768-e914db00-9bb7-11ea-8326-eb7214b7f023.png)
![image](https://user-images.githubusercontent.com/8146344/82606751-e1553680-9bb7-11ea-9f1e-38ba5e43bda2.png)

**Comment by iliakan** [comment]
I updated it once again.

Try `./edit ar` - you should see green body:

![image](https://user-images.githubusercontent.com/349336/82608462-685bdc00-9bc3-11ea-9ab2-4c3bfb50bbe9.png)

**Comment by OmarioHasan** [comment]
@iliakan  It's not working 
![image](https://user-images.githubusercontent.com/8146344/82609386-bf11e780-9bbc-11ea-979d-c7a86225dcd0.png)
![image](https://user-images.githubusercontent.com/8146344/82609407-cafda980-9bbc-11ea-9a16-147275c804a7.png)

**Comment by iliakan** [comment]
You do it, and then re-run ./edit ar?

Also, do you clear browser cache?


---
Best Regards,
Ilya Kantor
https://javascript.info

**Comment by OmarioHasan** [comment]
Yes I'm closed and reserving about 5 times and I'm serving on incognito window

**Comment by iliakan** [comment]
So my change isn't working for you? 

Here's how it should look:
![image](https://user-images.githubusercontent.com/349336/82610143-870fa200-9bc6-11ea-919a-08aee739907c.png)

Please note: incognito doesn't mean "no cache". It merely hides your identifying tokens/empties the environment. So you may still have it cached. I guess, that's the reason.

Also, when you see things working, you can try `./dev ar`

**Comment by iliakan** [comment]
Please check `public/en/pack/styles.css`, are the changes there, in the `body{...}` clause?

**Comment by OmarioHasan** [comment]
I got this error when using `./dev ar`

![image](https://user-images.githubusercontent.com/8146344/82611085-fdf56c80-9bbf-11ea-9347-56aeaafd9bbd.png)

**Comment by iliakan** [comment]
Yes, `./dev` expects to see `ar.yml` files, the translation of the web interface. That's not needed for `./edit` that uses `en.yml` by default.

Note: rtl.styl is not used now. There should be `body{...}` style with the change.

**Comment by OmarioHasan** [comment]
I'm using this file `server\modules\styles\blocks\body\body.styl` and this is what it includes

![image](https://user-images.githubusercontent.com/8146344/82611339-94299280-9bc0-11ea-9bdb-8b70456ecad6.png)

**Comment by OmarioHasan** [comment]
i think it reads rtl as variable 

![image](https://user-images.githubusercontent.com/8146344/82612083-50378d00-9bc2-11ea-82a2-c339aa4952b1.png)

**Comment by OmarioHasan** [comment]
Here is the problem

![image](https://user-images.githubusercontent.com/8146344/82612810-0e0f4b00-9bc4-11ea-9e4e-51122c07d275.png)

**Comment by iliakan** [comment]
Strange, please `git pull` latest?

Here's my `edit`:
```
/js/server master> cat edit
#!/usr/bin/env bash

# ./edit ru <- tutorial language: RU, server: EN
# ./edit ru ru <- tutorial language: RU, server: RU

: ${1?"Usage: $0 <tutorial language> [<server language>]"}


export TUTORIAL_ROOT="../$1.javascript.info"
export NODE_LANG="${2:-en}"
export TUTORIAL_LANG=$1 <-------------------------
```

So when you run `./edit ar`, the variable should be there.

**Comment by OmarioHasan** [comment]
It's the latest already

![image](https://user-images.githubusercontent.com/8146344/82612991-9130a100-9bc4-11ea-9ac2-6e3a5f99cc06.png)

**Comment by iliakan** [comment]
That's odd indeed. Why the variable does not get through then?

**Comment by OmarioHasan** [comment]
I'm using the engine and server and ar projects through the fork,
can this make any problems?

**Comment by iliakan** [comment]
If you pull from upstream, so that the fork is up to date, then it shouldn't be the reason.

**Comment by OmarioHasan** [comment]
did you recommend to remove the local and fork projects and repeat this process again?

**Comment by iliakan** [comment]
The `edit` file is correct: there's the variable `TUTORIAL_LANG`. But then it must be in `process.env.TUTORIAL_LANG`. That has nothing to do with my scripts, it's a straightforward node.js feature.

**Comment by OmarioHasan** [comment]
I think i've found what we can fix 

![image](https://user-images.githubusercontent.com/8146344/82614427-1f5a5680-9bc8-11ea-8d60-c97a69ac0d9e.png)

![image](https://user-images.githubusercontent.com/8146344/82614493-4dd83180-9bc8-11ea-8726-6f4f0915fabe.png)

![image](https://user-images.githubusercontent.com/8146344/82614638-a0b1e900-9bc8-11ea-949c-38ff01a052ea.png)


![image](https://user-images.githubusercontent.com/8146344/82614623-9263cd00-9bc8-11ea-87a1-2bf072a97427.png)

**Comment by iliakan** [comment]
Of course, that explains everything. I assumed you're running `./edit`, not `./edit.cmd`. Great, so now it works!

**Comment by OmarioHasan** [comment]
@iliakan  I think there is a problem related to displaying SVG
is that just a local issue for me?

![image](https://user-images.githubusercontent.com/8146344/82658055-ad652a00-9c26-11ea-9687-e805039d8b5d.png)

**Comment by iliakan** [comment]
Do you have ImageMagick installed? It tries to get image size.

**Comment by iliakan** [comment]
It might be better to make separate issues for different stuff, not to write everything here, in the progress tracker.

**Comment by OmarioHasan** [comment]
ok no problem but did we need to install ImageMagick

**Comment by iliakan** [comment]
No need, unless you want to see pictures locally.

**Comment by OmarioHasan** [comment]
I also made a PR for RTL in server hope changes merged to use it with my team

**Comment by iliakan** [comment]
I'll merge it later, after you evaluate it during the translation, ok? Anyway, RTL isn't yet used, as no translation exists.

**Comment by OmarioHasan** [comment]
My Team about 18 developers is using this style and they have now yesterday version with background green

**Comment by OmarioHasan** [comment]
at least this PR

**Comment by iliakan** [comment]
Yes, green was temporary =) merged it.

**Comment by dvmhmdsd** [comment]
Symbol type

**Comment by OmarioHasan** [comment]
> Symbol type

التوبيك ده فى حد هيشتغل فيه فعلا ممكن تختار توبيكس من هنا https://docs.google.com/spreadsheets/d/1axChrzLm-DLpAG9OjoWzBz9hu0sfxttlQ9GyhsegZ4E/edit#gid=0

**Comment by dvmhmdsd** [comment]
> التوبيك ده فى حد هيشتغل فيه فعلا ممكن تختار توبيكس من هنا

أنا اللى كاتب إسمي

**Comment by OmarioHasan** [comment]
Manuals and specifications

**Comment by Eyad-Mohammed-Osama** [comment]
An Introduction to JavaScript

**Comment by dvmhmdsd** [comment]
Object to primitive conversion

**Comment by Salah856** [comment]
Comparisons

**Comment by Salah856** [comment]
Basic operators, maths

**Comment by dvmhmdsd** [comment]
Iterables

**Comment by OmarioHasan** [comment]
Property getters and setters

**Comment by Salah856** [comment]
Attributes and properties

**Comment by OmarioHasan** [comment]
Property flags and descriptors

**Comment by Salah856** [comment]
Node properties: type, tag and contents

**Comment by dvmhmdsd** [comment]
JSON methods, toJSON

**Comment by dvmhmdsd** [comment]
Native prototypes

**Comment by dvmhmdsd** [comment]
Prototype methods, objects without proto

**Comment by dvmhmdsd** [comment]
Prototype methods, objects without **proto**

**Comment by Salah856** [comment]
Coordinates

**Comment by IEEE-AlAzhar** [comment]
Generators

**Comment by Salah856** [comment]
Methods of RegExp and String

**Comment by OmarioHasan** [comment]
@iliakan  What is the less percentage that we can go live with AR translation,
I see that the Turkish is live and it has 42%,
this will help me encourage the team and increase our speed

**Comment by iliakan** [comment]
@OmarioHasan Turkish had >50% when it became alive, now 42%, because the English version was updated since then.

I believe, at least one part of the tutorial must be ready to publish. The more the better =)

**Comment by IEEE-AlAzhar** [comment]
Async iterators and generators

**Comment by dvmhmdsd** [comment]
Proxy and Reflect

**Comment by Salah856** [comment]
Sticky flag "y", searching at position

**Comment by Salah856** [comment]
Catastrophic backtracking

**Comment by Salah856** [comment]
Backreferences in pattern: \N and \k

**Comment by Salah856** [comment]
Alternation (OR) |

**Comment by Salah856** [comment]
Class basic syntax

**Comment by Salah856** [comment]
Class basic syntax

**Comment by Salah856** [comment]
Class inheritance

**Comment by Salah856** [comment]
Lookahead and lookbehind

**Comment by Salah856** [comment]
Lookahead and lookbehind

**Comment by OmarioHasan** [comment]
Mixins

**Comment by Salah856** [comment]
Static properties and methods

**Comment by Salah856** [comment]
Extending built-in classes

**Comment by moaztayea** [comment]
Methods of primitives

**Comment by moaztayea** [comment]
Numbers

**Comment by Salah856** [comment]
Private and protected properties and methods

**Comment by Salah856** [comment]
Error handling with promises

**Comment by Salah856** [comment]
Microtasks

**Comment by OmarioHasan** [comment]
Polyfills

**Comment by Salah856** [comment]
Comments

**Comment by Salah856** [comment]
Promise API

**Comment by Salah856** [comment]
Async/await

**Comment by Salah856** [comment]
Capturing groups

**Comment by Salah856** [comment]
Window sizes and scrolling

**Comment by Salah856** [comment]
Element size and scrolling

**Comment by Salah856** [comment]
Modifying the document

**Comment by Adham-Niazy** [comment]
Recursion and stack

**Comment by MustafaMadkour** [comment]
Prototypal inheritance

**Comment by Adham-Niazy** [comment]
Function object, NFE

**Comment by MostafaMahmoud93** [comment]
BigInt

**Comment by Salah856** [comment]
Dynamic imports

**Comment by abdelrhmanali29** [comment]
Browser environment, specs

**Comment by Salah856** [comment]
Modules, introduction

**Comment by Salah856** [comment]
Export and Import

**Comment by Salah856** [comment]
Map and Set

**Comment by Salah856** [comment]
Date and time

**Comment by Salah856** [comment]
Destructuring assignment

**Comment by Salah856** [comment]
Map and Set

**Comment by Salah856** [comment]
Map and Set

**Comment by Salah856** [comment]
Developer console

**Comment by Salah856** [comment]
The modern mode, "use strict"

**Comment by Salah856** [comment]
Eval: run a code string

**Comment by Salah856** [comment]
Currying

**Comment by atefBB** [comment]
<div dir="rtl">عمل ممتاز @Salah856 ! راجع تعليقاتي على عملك، مشكورا.</div>

**Comment by Salah856** [comment]
Arrow functions revisited

**Comment by Salah856** [comment]
Function binding

**Comment by Salah856** [comment]
Decorators and forwarding, call/apply

**Comment by Salah856** [comment]
Scheduling: setTimeout and setInterval

**Comment by OmarioHasan** [comment]
Reference Type

**Comment by Sara-Elsayed** [comment]
Objects

**Comment by Salah856** [comment]
F.prototype

**Comment by Ahmed-Adel3** [comment]
Object methods, "this"

**Comment by Abdelrahmansherwida** [comment]
Array methods

**Comment by Adham-Niazy** [comment]
Rest parameters and spread syntax

**Comment by Abdelrahmansherwida** [comment]
Array methods

**Comment by AhmedIbrahimGalal** [comment]
Constructor, operator "new"

**Comment by Ahmed-Adel3** [comment]
Prototypal inheritance

**Comment by Adham-Niazy** [comment]
The old "var"

**Comment by mohamedElkast** [comment]
Strings

**Comment by Adham-Niazy** [comment]
Global object

**Comment by AhmedIbrahimGalal** [comment]
Promise

**Comment by AhmedIbrahimGalal** [comment]
Promise

**Comment by AhmedIbrahimGalal** [comment]
promise-basics

**Comment by AhmedIbrahimGalal** [comment]
Promise

**Comment by AhmedIbrahimGalal** [comment]
Promises chaining

**Comment by AhmedIbrahimGalal** [comment]
Code structure

**Comment by Abdelrahmansherwida** [comment]
Automated testing with Mocha

**Comment by Adham-Niazy** [comment]
Strings

**Comment by mohamedElkast** [comment]
Variable scope

**Comment by AhmedIbrahimGalal** [comment]
Custom errors, extending Error

**Comment by Ahmed-Adel3** [comment]
Class checking: "instanceof"

**Comment by Abdelrahmansherwida** [comment]
Object.keys, values, entries

**Comment by Ahmed-Adel3** [comment]
Numbers

**Comment by Ahmed-Adel3** [comment]
Numbers

**Comment by Adham-Niazy** [comment]
Variable scope

**Comment by Ahmed-Adel3** [comment]
Object methods, "this"

**Comment by belalmosad** [comment]
Introduction to browser events

**Comment by belalmosad** [comment]
Hi, it's my first contribution here, i'm facing a problem. There is a conflict between the Arabic text and English text, like in the picture: 
![image](https://user-images.githubusercontent.com/45633405/86525452-7fab0c00-be87-11ea-906f-cb1efa3a57c6.png)
how to handle such problem? Thanks.

**Comment by abdelrhmanali29** [comment]
Styles and classes

**Comment by ahmedswaby** [comment]
Events: change, input, cut, copy, paste

**Comment by medaminefh** [comment]
An Introduction to JavaScript

**Comment by AbderrazzakB** [comment]
Walking the DOM

**Comment by Abdelwahab07** [comment]
Event delegation

**Comment by MoayadShalish** [comment]
Fetch

**Comment by ibrahim-301** [comment]
LocalStorage, sessionStorage

**Comment by mahmoudmagdy7** [comment]
Fetch

**Comment by mahmoudmagdy7** [comment]
XMLHttpRequest

**Comment by afilahkle** [comment]
Event delegation

**Comment by afilahkle** [comment]
Nullish coalescing operator '??'

**Comment by afilahkle** [comment]
Polyfills and transpilers

**Comment by afilahkle** [comment]
Object references and copying

**Comment by afilahkle** [comment]
Variable scope, closure

**Comment by afilahkle** [comment]
Error handling, "try...catch"

**Comment by afilahkle** [comment]
Async iteration and generators

**Comment by aladdinshammr** [comment]
Searching: getElement*, querySelector*

**Comment by bakiroo** [comment]
Hello, <em>nice project!</em> and I like to be a part as contributor, who can add more value. My Arabic language skill is <strong>Strong</strong>, JS skill not yet I'm working hard to improving it, but I'm familly with CODE. <em>Thank's to all contribitors</em>.
